mesa_expense_autopopulate/models/hr/expense.py

```diff
@@ -0,0 +1,132 @@
+from odoo import api, fields, models
+
+
+class Expense(models.Model):
+    _inherit = "hr.expense"
+
+    x_studio_field_Evfct = fields.Many2one(
+        "project.project",
+        string="Project",
+        compute="_compute_project",
+        inverse="_inverse_project",
+    )
+    team_id = fields.Many2one(compute="_compute_team", inverse="_inverse_team")
+    warehouse_id = fields.Many2one(
+        compute="_compute_warehouse", inverse="_inverse_warehouse"
+    )
+    user_id = fields.Many2one(compute="_compute_user", inverse="_inverse_user")
+    company_id = fields.Many2one(compute="_compute_company", inverse="_inverse_company")
+    # The following studio field is only included here because the database would otherwise not let this module install.
+    x_studio_aa_update = fields.Many2one("account.analytic.account")
+
+    @api.multi
+    @api.depends("sale_order_id")
```

Comment:
```suggestion
    @api.depends("sale_order_id.x_studio_project")
```
You should probably include the field you're actually using in your dependency list.  Like if someone were to change the project on the sale order, Odoo wouldn't recompute the field.

Reply:
The same goes for all of the `compute`s below.